test/service.spec.js

```diff
@@ -1,16 +1,73 @@
 // Generated by CoffeeScript 1.12.3
 (function() {
-  var AWS, MockService, helpers, metadata,
+  var AWS, MockService, helpers, metadata, MockServiceWithErrorCodeMapping
     hasProp = {}.hasOwnProperty;
 
   helpers = require('./helpers');
 
   AWS = helpers.AWS;
 
   MockService = helpers.MockService;
+  MockServiceWithErrorCodeMapping = helpers.MockServiceWithErrorCodeMapping;
 
   metadata = require('../apis/metadata.json');
 
+  describe('AWSQuery compatible error codes', function () {
+    let service;
+    let mapping = { ServiceError: { code: 'Custom.ErrorCode' } };
+
+    beforeEach(function(done) {
+      let CustomService = MockServiceWithErrorCodeMapping(mapping);
```

Comment:
Can you use `MockServiceFromApi()` instead?

Reply:
See comment above